src/azure-cli-core/azure/cli/core/_msal.py

```diff
@@ -19,16 +19,14 @@ def _acquire_token_silent_by_finding_rt_belongs_to_me_or_my_family(
 
     def _acquire_token_silent_by_finding_specific_refresh_token(
             self, authority, scopes, query,
-            rt_remover=None, break_condition=lambda response: False, **kwargs):
+            rt_remover=None, break_condition=lambda response: False,
+            force_refresh=False, correlation_id=None, claims_challenge=None, **kwargs):
         refresh_token = kwargs.get('refresh_token', None)
         client = self._build_client(self.client_credential, authority)
         if 'refresh_token' in kwargs:
             kwargs.pop('refresh_token')
-        if 'force_refresh' in kwargs:
-            kwargs.pop('force_refresh')
-        if 'correlation_id' in kwargs:
-            kwargs.pop('correlation_id')
-        response = client.obtain_token_by_refresh_token(refresh_token, scope=scopes, **kwargs)
+        response = client.obtain_token_by_refresh_token(
+            {'refresh_token': refresh_token, 'credential_type': 'RefreshToken'}, scope=scopes, **kwargs)
```

Comment:
We can directly use `refresh_token` as `token_item` in `str` for simplicity:

https://github.com/AzureAD/microsoft-authentication-library-for-python/blob/ff7629b38ea873b0a8d3609ce0e04224a84a147e/msal/oauth2cli/oauth2.py#L494-L506

```py
    def obtain_token_by_refresh_token(self, token_item, scope=None,
            rt_getter=lambda token_item: token_item["refresh_token"],
            on_removing_rt=None,
            on_updating_rt=None,
            **kwargs):
        ...
        :param token_item:
            A refresh token (RT) item, in flexible format. It can be a string,
            or a whatever data structure containing RT string and its metadata,
            in such case the `rt_getter` callable must be able to
            extract the RT string out from the token item data structure.
```

Reply:
Use this hack due to https://github.com/AzureAD/microsoft-authentication-library-for-python/issues/297